Automattic/wp-calypso
# codeowners: Add I18n section

#### Changes proposed in this Pull Request

* Add I18n section with several paths that have been assigned to @Automattic/i18n as code owner.

#### Testing instructions

* Review proposed paths and confirm they've been properly assigned to @Automattic/i18n.
* Check if there are any other paths that could be added to the list.
